core/lib/types/src/api/mod.rs

```diff
@@ -209,7 +209,7 @@ pub struct TransactionReceipt {
     pub transaction_index: Index,
     /// Hash of the block this transaction was included within.
     #[serde(rename = "blockHash")]
-    pub block_hash: Option<H256>,
+    pub block_hash: H256,
```

Comment:
If you're changing the struct then it also makes sense to make `block_number`, `status` required